File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.java
class Solution {
    public int beautySum(String s) {
        int str_length = s.length();
        int total = 0;
        for(int i=0;i<str_length;i++){
            int [] counter = new int[26];
            for(int j =i;j<str_length;j++){
                counter[s.charAt(j)-'a']++;

                int maxFrequency = 0;
                int minFrequency = 1000;
                for (int freq : counter){
                    if(freq>0){
                         maxFrequency = Math.max(freq,maxFrequency);
                        minFrequency = Math.min(freq,minFrequency);
                    }
                   
                }
                total += maxFrequency-minFrequency;
            }
        }return total;
    }
}

File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.py
from collections import Counter
class Solution:
    def beautySum(self, s: str) -> int:
        total = 0
        str_length = len(s)
        for i in range(str_length):
            char_counter = Counter()
            for j in range(i,str_length):
                char_counter[s[j]]+=1
                curr_beauty = max(char_counter.values())-min(char_counter.values())
                total += curr_beauty
        return total